Realm/RLMRealm.mm

```diff
@@ -289,6 +290,12 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
     configuration = [configuration copy];
     Realm::Config& config = configuration.config;
 
+    if (configuration.encryptionKey && config.sync_config) {
+        std::array<char, 64> sync_encryption_key;
+        std::copy_n(config.encryption_key.begin(), 64, sync_encryption_key.begin());
+        config.sync_config.get()->realm_encryption_key = std::move(sync_encryption_key);
```

Comment:
Having a weird special case in some unrelated code is much worse than an extra few lines of code.